src/libstore/globals.cc

```diff
@@ -252,6 +254,47 @@ void PluginFilesSetting::set(const std::string & str, bool append)
 }
 
 
+void EnvironmentSetting::set(const std::string & str, bool append)
+{
+    if (! append) {
+        for (auto & [name, oldValue] : oldEnvironment) {
+            if (oldValue.has_value()) {
+                setenv(name.c_str(), oldValue->c_str(), 1);
+            } else {
+                unsetenv(name.c_str());
+            }
+        }
+        oldEnvironment.clear();
+        value.clear();
+        hasInherited = false;
+    }
+
+    StringMap newValues;
+
+    for (auto & elem : tokenizeString<Strings>(str)) {
+        size_t pos = elem.find('=');
+        if (pos == std::string::npos) {
+            // name
+            char * value = getenv(elem.c_str());
```

Comment:
Nix has its own `getenv` wrapper to directly manipulate c++ strings:

```suggestion
            std::optional<std::string> value = getEnv(elem);
```

(and changing the uses below)

Reply:
As you said :ok_hand: